thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/dashboard/resources/OnboardResource.java

```diff
@@ -217,4 +222,100 @@ public Boolean cloneAnomalyInstances(Long srcId, Long destId) {
+      @QueryParam("monitoringWindowEndTime") long monitoringWindowEndTime) {
+    LOG.info("Deleting anomalies in the time range: {} -- {}", new DateTime(monitoringWindowStartTime), new
+        DateTime(monitoringWindowEndTime));
+
+    long functionId = Long.valueOf(id);
+
+    int mergedAnomaliesDeleted = 0;
+    AnomalyFunctionDTO anomalyFunction = anomalyFunctionDAO.findById(functionId);
+    LOG.info("Beginning cleanup of functionId {} collection {} metric {}",
+        functionId, anomalyFunction.getCollection(), anomalyFunction.getMetric());
+    // Find merged anomalies and delete them first
+    List<MergedAnomalyResultDTO> mergedResults =
+        mergedAnomalyResultDAO.findByStartTimeInRangeAndFunctionId(monitoringWindowStartTime, monitoringWindowEndTime, functionId);
+    if (CollectionUtils.isNotEmpty(mergedResults)) {
+      mergedAnomaliesDeleted = deleteMergedResults(mergedResults);
+    }
+
+    LOG.info("Deleted {} merged anomalies", mergedAnomaliesDeleted);
+    return mergedAnomaliesDeleted;
+  }
+
+  // Delete merged anomalies from mergedAnomalyResultDAO
+  private int deleteMergedResults(List<MergedAnomalyResultDTO> mergedResults) {
+    LOG.info("Deleting merged results");
+    int mergedAnomaliesDeleted = 0;
+    for (MergedAnomalyResultDTO mergedResult : mergedResults) {
+      // Delete raw anomalies of the merged anomaly
+      List<RawAnomalyResultDTO> rawAnomalyResultDTOs = mergedResult.getAnomalyResults();
+      //deleteRawResults(rawAnomalyResultDTOs);
+
+      LOG.info(".....Deleting id {} for functionId {}", mergedResult.getId(), mergedResult.getFunctionId());
+      mergedAnomalyResultDAO.delete(mergedResult);
+      mergedAnomaliesDeleted++;
+    }
+    return mergedAnomaliesDeleted;
+  }
+
+  /**
+   * Breaks down the given range into consecutive monitoring windows as per function definition
+   * Regenerates anomalies for each window separately
+   * @param id anomaly function id
+   * @param detectionHost the host of anomaly detection function
+   * @param detectionPort the port of anomaly detection function
+   * @param monitoringWindowStartTime The start time of the monitoring window (in milli-second)
+   * @param monitoringWindowEndTime The start time of the monitoring window (in milli-second)
+   * @param isForceBackfill determine if it should backfill
+   * @return
+   * @throws Exception
+   */
+  @POST
+  @Path("function/{id}/regenerateAnomaliesInRange")
+  public String regenerateAnomaliesInRange(@PathParam("id") String id,
+      @QueryParam("detectionHost") String detectionHost,
+      @QueryParam("detectionHost") int detectionPort,
+      @QueryParam("monitoringWindowStartTime") long monitoringWindowStartTime,
+      @QueryParam("monitoringWindowEndTime") long monitoringWindowEndTime,
+      @QueryParam("isForceBackfill") @DefaultValue("false") String isForceBackfill) throws Exception {
+    long functionId = Long.valueOf(id);
```

Comment:
should not this take care of deleting the existing anomalies for this time window first before re - generating ?